src/components/tvShow/tvShow.js

```diff
@@ -1,78 +1,110 @@
 import React, { Component } from 'react';
+import { Route } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import showRepository from '../../repository/tvShowRepository';
-import Spinner from '../ui/spinner';
 import { FETCH_STATES } from '../constants';
+import TVShowMenu from './tvShowMenu';
+import TVShowInfo from './tvShowInfo';
+import Spinner from '../ui/spinner';
+import TVShowSeasons from './tvShowSeasons';
 
 class TVShow extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      fetchState: null,
-      show: null,
+      show: {
+        fetchState: null,
+        data: null,
+      },
+      seasons: {
+        fetchState: null,
+        data: null,
+      },
     };
 
-    this.onClick = this.onClick.bind(this);
+    this.toggleFavourite = this.toggleFavourite.bind(this);
   }
 
   componentWillMount() {
     const showId = parseInt(this.props.match.params.showId, 10);
-    this.setState({ fetchState: FETCH_STATES.PENDING });
+
+    this.setState({ show: { fetchState: FETCH_STATES.PENDING } });
     showRepository.findById(showId)
-      .then(show => this.setState({ show, fetchState: FETCH_STATES.SUCCESS }))
+      .then(data => this.setState({ show: { data, fetchState: FETCH_STATES.SUCCESS } }))
       .catch((err) => {
         const fetchState = (err.message === 'Not Found') ? FETCH_STATES.SUCCESS : FETCH_STATES.FAILED;
-        this.setState({ fetchState });
+        this.setState({ show: { fetchState } });
+      });
+
+    this.setState({ seasons: { fetchState: FETCH_STATES.PENDING } });
+    showRepository.findSeasonsWithEpisodes(showId)
+      .then(data => this.setState({ seasons: { data, fetchState: FETCH_STATES.SUCCESS } }))
+      .catch((err) => {
+        const fetchState = (err.message === 'Not Found') ? FETCH_STATES.SUCCESS : FETCH_STATES.FAILED;
+        this.setState({ seasons: { fetchState } });
       });
   }
 
-  onClick = tvShowId => (e) => {
+  toggleFavourite = tvShowId => (e) => {
     e.preventDefault();
     this.props.toggleFavourite(tvShowId);
   };
 
-  renderShow() {
-    const { show, fetchState } = this.state;
-    const { isFavourite } = this.props;
-    const favImage = isFavourite ? 'star-filled' : 'star-unfilled';
-    const favAlt = isFavourite ? 'Filled star' : 'Unfilled star';
+  renderShowContent = () => {
+    const { isFavourite, match } = this.props;
+    const { data: show = null, fetchState: showFetchState } = this.state.show;
+    const { data: seasons = [], fetchState: seasonsFetchState } = this.state.seasons;
+
+    if (showFetchState === FETCH_STATES.PENDING) {
+      return <Spinner visible={showFetchState === FETCH_STATES.PENDING} />;
```

Comment:
jeśli showFetchState === PENDING to linijke niżej nie musisz podawać `showFetchState === FETCH_STATES.PENDING` bo i tak wiesz że to wyniesie `true` ;)